tests/models/beit/test_modeling_beit.py

```diff
@@ -64,7 +67,7 @@ def __init__(
         is_training=True,
         use_labels=True,
         hidden_size=32,
-        num_hidden_layers=2,
+        num_hidden_layers=4,
```

Comment:
The semantic segmentation head has a lot of hardcoded values which assume 4 backbone feature maps, such as [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/beit/modeling_beit.py#L1153-L1164) and [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/beit/modeling_beit.py#L1008). Hence if you only use 2 hidden layers, the model will complain.

The reason this still passed on main was because in the current implementation, `hidden_states` also includes the initial embeddings, hence `hidden_states` will be a list of length 3. I see that if I set `num_hidden_layers=3`, the tests also pass.

However, that means we would have to remove the `ValueError` which is raised which notifies the user that one should use 4 feature maps for the backbone, since the head assumes this.

Reply:
Agreed - let's keep it as 4 here :)

Reply:
The reason it passes with num_hidden_layers=3 is because in that case the first 3 FPNs will be taken [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/beit/modeling_beit.py#L1254-L1255). There are 3 lateral convs added here, so this allows the model to still perform a forward pass. 3 is basically the minimum of feature maps you need to get it to work.